x-pack/plugins/fleet/server/services/security/uninstall_token_service/index.ts

```diff
@@ -61,6 +62,14 @@ interface UninstallTokenSOAggregation {
   by_policy_id: AggregationsMultiBucketAggregateBase<UninstallTokenSOAggregationBucket>;
 }
 
+function getNamespaceFiltering(namespace: string) {
+  if (namespace === DEFAULT_NAMESPACE_STRING) {
+    return `(${UNINSTALL_TOKENS_SAVED_OBJECT_TYPE}.attributes.namespaces:default) or (not ${UNINSTALL_TOKENS_SAVED_OBJECT_TYPE}.attributes.namespaces:*)`;
```

Comment:
Just a clarification, is the condition after the `or` needed to filter out those having `namespace = *`? I'm not totally sure about this condition, what is going to match.

Reply:
if we want the token in the default space, we need to filter by object either not having `namespaces` set or where `namespaces` include `default, does it make sense to you?

Reply:
isn't going to match also everything else that has any namespace set because of the wildcard `*`? But maybe I'm misunderstanding how this works.

Reply:
the condition is `not namespaces:*`  easy to miss the `not` here